### PR TITLE
[Feature] Reusable Card Component

### PR DESCRIPTION
**Pull request by gertie-sheshe**
## Description

Reusable Card Component

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## Screenshots
screencapture-localhost-3000-2021-08-05-14_45_05

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/components/Card/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          {image && (
+            <div className={classes.cardMedia}>
+              <Image src={image} layout="fill" className={classes.image} />
+            </div>
+          )}
+          <CardContent className={classes.content}>{content}</CardContent>
+        </CardActionArea>
+      ) : (
+        <CardContent className={classes.content}>
+          {image && (
```
Comment (kilemensi) [inline src/components/Card/index.js:35]: Shouldn't `image` be in `CardContent` in both/neither situations?

File: src/components/Card/useStyles.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(({ breakpoints, typography }) => ({
+  root: {},
+  cardMedia: {
+    // TODO(kilemensi): Confirm if all image will have invisible margins
+    //                  around them. If yes, leave this as is. Otherwise
+    //                  remove all negative margins calculations
```
Comment (kilemensi) [inline src/components/Card/useStyles.js:8]: Have we confirmed if all images will have margins?

File: src/components/Card/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            <div className={classes.cardMedia}>
+              <Image src={image} layout="fill" className={classes.image} />
+            </div>
+          )}
+          {content}
```
Comment (kilemensi) [inline src/components/Card/index.js:40]: Why not `children`?

**Review by kilemensi — COMMENTED** [review]
👍🏾

File: src/components/Card/index.js (hunk 1 above)
Comment (gertie-sheshe) [reply src/components/Card/index.js]: Fixed

File: src/components/Card/index.js (hunk 3 above)
Comment (gertie-sheshe) [reply src/components/Card/index.js]: Fixed

File: src/components/Card/useStyles.js (hunk 2 above)
Comment (gertie-sheshe) [reply src/components/Card/useStyles.js]: Seems only the Explore images have the extra margins. The rest of the images are fine.

**Review by kilemensi — APPROVED** [review]
> Seems only the Explore images have the extra margins. The rest of the images are fine.

If the margin styling doesn't affect the other cards, lets update that comment then to say it's for explore card only, etc.

File: src/components/InsightsData/index.js [hunk 4]
```diff
@@ -24,7 +24,13 @@ const InsightsData = ({ title, subtitle, items, ...props }) => {
         <RichTypography className={classes.subtitle}>{subtitle}</RichTypography>
         <Grid container className={classes.list}>
           {itemsData &&
-            itemsData.map((item) => <InsightCard key={item.title} {...item} />)}
+            itemsData.map((item) => {
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/InsightsData/index.js]: Fixed

File: src/components/Card/useStyles.js (hunk 2 above)
Comment (KhadijaMahanga) [reply src/components/Card/useStyles.js]: @gertie-sheshe I should have tagged you on [this](https://trello.com/c/sznZ9PbN/43-xd-links-for-devs) Images have box-shadows and not margins cc @kilemensi

File: src/components/ExploreCard/useStyles.js [hunk 5]
```diff
@@ -2,7 +2,7 @@ import { makeStyles } from "@material-ui/core/styles";
 
 const useStyles = makeStyles(({ breakpoints, typography }) => ({
   root: {
-    minWidth: typography.pxToRem(350),
+    maxWidth: typography.pxToRem(350),
```
Comment (KhadijaMahanga) [inline src/components/ExploreCard/useStyles.js:5]: same same

File: src/components/InsightCard/useStyles.js [hunk 6]
```diff
@@ -31,22 +35,6 @@ const useStyles = makeStyles(({ typography, breakpoints }) => ({
   linkText: {
     fontWeight: "bold",
   },
-  image: {
-    objectFit: "contain",
-  },
```
Comment (KhadijaMahanga) [inline src/components/InsightCard/useStyles.js:36]: like this

**Comment by KhadijaMahanga** [comment]
@gertie-sheshe not related to this PR, but stories appear first before visual card on homepage.  Should be vice-versa

**Review by gertie-sheshe — COMMENTED** [review]
huh?

File: src/components/Card/useStyles.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        width: typography.pxToRem(376),
+      },
+    };
+  },
+  content: {
+    padding: 0,
+  },
+}));
+
+export default useStyles;
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/Card/useStyles.js:66]: Fixed

File: src/components/DataVisualCard/Content.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+Content.propTypes = {
+  description: PropTypes.string,
+};
+
+Content.defaultProps = {
+  description: undefined,
+};
+
+export default Content;
```
[the remark this answers is not in the document]
Comment (KhadijaMahanga) [reply src/components/DataVisualCard/Content.js:28]: @gertie-sheshe `RichTypography` is already an independent component/module. Putting in another component on itself is redundant..it's not modularizing

**Review by KhadijaMahanga — APPROVED** [review]

**Review by nerdlet — APPROVED** [review]
@gertie-sheshe Looks good 🚀 Reminder to update the PR template and add screenshots when possible

**Pull request merged**